src/Zenaton/v1/Client.php

```diff
@@ -249,6 +252,146 @@ public function resumeWorkflow($workflowName, $customId)
         $this->updateInstance($workflowName, $customId, self::WORKFLOW_RUN);
     }
 
+    /**
+     * Schedule a task instance.
+     *
+     * @param string $cron
+     *
+     * @throws ApiException
+     *
+     * @return Schedule
+     */
+    public function scheduleTask(TaskInterface $task, $cron)
+    {
+        $name = \get_class($task);
+
+        $mutation = <<<'MUTATION'
+            mutation createTaskSchedule($input: CreateTaskScheduleInput!) {
+                createTaskSchedule(input: $input) {
+                    schedule {
+                        cron
+                        id
+                        name
+                        target {
+                            ... on TaskTarget {
+                                codePathVersion
+                                initialLibraryVersion
+                                name
+                                programmingLanguage
+                                properties
+                                type
+                            }
+                        }
+                        insertedAt
+                        updatedAt
+                    }
+                }
+            }
+MUTATION;
+
+        $variables = [
+            'input' => [
+                'environmentName' => $this->appEnv,
+                'cron' => $cron,
+                'intentId' => $this->uuidFactory->uuid4()->toString(),
+                'programmingLanguage' => static::PROG,
+                'properties' => $this->serializer->encode($this->properties->getPropertiesFromObject($task)),
+                'taskName' => $name,
+            ],
+        ];
+
+        try {
```

Comment:
Would it make sense to extract this `try` block into a tiny graphql client module? Especially if we are to use more of alfred in the future.

Reply:
Yes. The logic is always the  same:
- try
- catch ConnectionErrorException and throw an ApiException instead.
- check if an error list is in the response body to build an ApiException if needed.

Otherwise, I can do what I need to do with the response body. I agree we can extract what I listed in a small GraphQL class but I would like to wait for more use case in order to correctly design this client.
Removal of agent client mode is coming soon (the PR is already opened for PHP actually), which will bring a broader usage of Alfred in the libraries. I think this will be a good time to introduce it.